components/header/header.tsx

```diff
@@ -0,0 +1,72 @@
+'use client';
+
+import Image from 'next/image';
+import Link from 'next/link';
+import { useEffect, useState } from 'react';
+
+import { Button } from '../ui/button';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '../ui/dropdown-menu';
+
+export const Header = () => {
+  const [isSticky, setIsSticky] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const offset = window.scrollY;
+      setIsSticky(offset > 10);
+    };
+
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  return (
+    <header
+      className={`fixed top-0 z-50 flex w-full justify-between transition-all duration-300 ${
+        isSticky ? 'h-12 bg-orange-300 py-2 shadow-lg' : 'h-16 bg-transparent py-4'
+      }`}
```

Comment:
Лучше вынести классы в `cn (clsx + twMerge)` для читаемости и без конфликтов. Функцию`cn` я видела в `lib/utils.ts`
```suggestion
      className={cn(
  'fixed top-0 z-50 flex w-full justify-between transition-all duration-300',
  {
    'h-12 bg-orange-300 py-2 shadow-lg': isSticky,
    'h-16 bg-transparent py-4': !isSticky
  }
)}
```

Reply:
Сделала!